### PR TITLE
Shuffle host array at client init time

### LINKED ISSUE #75: Shuffle the default host arrays at client init time
The goal is to systematically avoid overloading the first non-DSN host upon failure. Shuffling should be done once and for all, as retrying a different host every time would likely imply opening a new connection, therefore incur a big overhead. Randomization across all users should be enough to guarantee proper load balancing.

### PR DESCRIPTION
**Pull request by clement-leprovost**
Fixes #75.

### CODE REVIEW

File: algoliasearch/src/main/java/com/algolia/search/saas/Client.java [hunk 1]
```diff
@@ -111,18 +113,26 @@ public Client(@NonNull String applicationID, @NonNull String apiKey, String[] ho
...
                     applicationID + "-1.algolianet.com",
                     applicationID + "-2.algolianet.com",
                     applicationID + "-3.algolianet.com"
             );
+            Collections.shuffle(fallbackHosts);
+            readHosts = new ArrayList<>(fallbackHosts.size() + 1);
+            readHosts.add(applicationID + "-dsn.algolia.net");
+            readHosts.addAll(fallbackHosts);
+            writeHosts = new ArrayList<>(fallbackHosts.size() + 1);
+            writeHosts.add(applicationID + ".algolia.net");
```
Comment (PLNech) [inline algoliasearch/src/main/java/com/algolia/search/saas/Client.java]: Why no `-dsn` on writeHosts?

Comment (ElPicador) [inline algoliasearch/src/main/java/com/algolia/search/saas/Client.java]: `-dsn` is only for the search.
write must be made to the cluster not the slaves

Comment (redox) [inline algoliasearch/src/main/java/com/algolia/search/saas/Client.java]: `appid-dsn` targets all up&running machines (master + slaves), `appid` targets all up&running machines accepting indexing tasks (master)

**Comment by PLNech** [comment]
LGTM :+1:

**Pull request merged**